app/reports/file_counts.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Generates a report of SDR objects, their types and the number of files they
+# contain.
+#
+# bin/rails r -e production "FileCounts.report"
+#
+# Or if you want to limit the results (e.g. top 100)
+#
+# bin/rails r -e production "FileCounts.report(100)"
+#
+class FileCounts
+  SQL = <<~SQL.squish.freeze
+    SELECT DISTINCT(external_identifier),
+      content_type,
+      JSONB_ARRAY_LENGTH(JSONB_PATH_QUERY(structural, '$.contains.structural.contains')) AS count
+    FROM dros
+    ORDER BY count DESC
+  SQL
+
+  def self.report(limit = nil)
+    sql = limit ? SQL + " LIMIT #{limit}" : SQL
```

Comment:
I think this is functionally equivalent and may be a smidge easier to (visually) parse?

```suggestion
  def self.report(limit = 'ALL')
    sql = "#{SQL} LIMIT #{limit}"
```

Reply:
I can try it -- I didn't think the `LIMIT` clause would work without an integer value?

Reply:
Oh it does have a value `ALL` -- I agree it looks simpler.